OSDOCS-3001: Added Day 2 information for proxies (#42320)
openshift/openshift-docs: modules/cluster-wide-proxy-updates.adoc

// Module included in the following assemblies:
//
// * networking/configuring-cluster-wide-proxy.adoc

:_content-type: PROCEDURE
[id="cluster-wide-proxy-updates_{context}"]

> = Updating your cluster-wide proxy

Reviewer: ```suggestion
= Configuring or updating your cluster-wide proxy after installation
```

> After configuring your cluster-wide proxy, you can perform additional configurations on the proxy or add an Additional Trust Bundle.

Reviewer: Sentence feels a little not right.. these steps are relevant if either:

- the cluster hasn't yet been configured with a proxy, but the cluster owner wants it to be
- the cluster has already been configured with a proxy, but now needs to be updated (for the reasons listed in the subsequent paragraph)

> You may need to perform these actions because the cluster-wide proxy was configured after installation, the network address needs to be updated, your Certificate Authorities expired and the Trust Bundle needs to be replaced.

Reviewer: ```suggestion
You may need to perform these actions if the cluster-wide proxy is being configured after installation, the proxy's network address needs to be updated, or if any of the proxy's Certificate Authorities have expired and the Additional Trust Bundle needs to be replaced.
```
Might read better as a bulleted list

[NOTE]
====
The cluster applies the configuration to the cluster’s control plane and worker nodes. This process results in each node in the cluster temporarily being placed into an unschedulable state and drained of its workloads while applying the configuration. Each node will be restarted as part of this process.
====

.Prerequsites

> * You have the `oc` CLI installed and configured.

Reviewer: Oh, I do see that `oc` commands are included in the subsequent "verification" steps. Maybe the `oc` pre-requisite can be moved down there.

.Procedure
* To create a cluster with a proxy, run the following command:
+
ifdef::openshift-rosa[]
[source,terminal]
----
$ rosa edit cluster \
 --cluster $CLUSTER_NAME \
 --additional-trust-bundle-file $CA_BUNDLE_FILE \ <1>
 --http-proxy $HTTP_PROXY \
 --https-proxy $HTTPS_PROXY
----
endif::[]
ifdef::openshift-dedicated[]
[source,terminal]
----

> $ rosa edit cluster \

Reviewer: ```suggestion
$ ocm edit cluster \
```

 --cluster $CLUSTER_NAME \
 --additional-trust-bundle-file $CA_BUNDLE_FILE \ <1>
 --http-proxy $HTTP_PROXY \
 --https-proxy $HTTPS_PROXY
----
endif::[]
+

> While the `additional-trust-bundle-file`, `http-proxy`, and `https-proxy` arguments are optional, if you set a `additional-trust-bundle-file` without either an `http-proxy` or `https-proxy` argument, then the additional trust bundle will not be configured to use a proxy.

Reviewer: I think we can probably leave this sentence out.. or reword it, though I'm not sure how to succinctly describe the effect that solely setting `additional-trust-bundle-file` has. Maybe change it to:

"then the additional trust bundle will still be used for verifying cluster system egress traffic."

* Verify that the configuration updates have been successfully applied:

> ** All of the MachineConfigPools are updated.

Reviewer: We can include an example here.

```
$ oc get machineconfigpools
```

// Module included in the following assemblies:
//
// * networking/configuring-cluster-wide-proxy.adoc

//OSDOCS-3001 Configuration for cluster-wide-proxy Day 2
:_content-type: PROCEDURE
[id="cluster-wide-proxy-updates_{context}"]
= Configuring or updating your cluster-wide proxy after installation

ifdef::openshift-rosa[]
As the cluster owner, you may wish to add a proxy to your created cluster after installation, or you may wish to make changes to your proxy that you configured during cluster installation. The `rosa` CLI provides some options for adding a proxy to your cluster or modifying an existing proxy on your cluster.
endif::[]
ifdef::openshift-dedicated[]
As the cluster owner, you may wish to add a proxy to your created cluster after installation, or you may wish to make changes to your proxy that you configured during cluster installation. The `ocm` CLI provides some options for adding a proxy to your cluster or modifying an existing proxy on your cluster.
endif::[]

You may need to perform these actions if:

* the cluster-wide proxy is configured after installation,

> * the proxy's network address needs to be updated, or

Reviewer 2: Maybe the 'or' is redundant

* You have the `ocm` CLI installed and configured.
endif::[]

.Procedure
* To edit a cluster, run the following command:
+
ifdef::openshift-rosa[]
[source,terminal]
----
$ rosa edit cluster \
 --cluster $CLUSTER_NAME \
 --additional-trust-bundle-file $CA_BUNDLE_FILE \
 --http-proxy $HTTP_PROXY \
 --https-proxy $HTTPS_PROXY
----
endif::[]
ifdef::openshift-dedicated[]
[source,terminal]
----
$ ocm edit cluster \
 --cluster $CLUSTER_NAME \
 --additional-trust-bundle-file $CA_BUNDLE_FILE \
 --http-proxy $HTTP_PROXY \
 --https-proxy $HTTPS_PROXY
----
endif::[]
+
While the `additional-trust-bundle-file`, `http-proxy`, and `https-proxy` arguments are optional, if you set a `additional-trust-bundle-file` without either an `http-proxy` or `https-proxy` argument, then the additional trust bundle will still be used for verifying cluster system egress traffic.
* You can verify that the proxy and certificate authority configuration updates have been successfully applied to your cluster by:
** All of the MachineConfigPools are updated. Run the following command to see their status:
+
[source,terminal]
----
$ oc get machineconfigpools
----
+
.Sample Output
[source,terminal]
----

> $ NEED EXAMPLE

Reviewer: Here is an example, though its quite wide!

The important thing in the example is the "Updated" column which says "true"

```
oc get machineconfigpools 
NAME     CONFIG                                             UPDATED   UPDATING   DEGRADED   MACHINECOUNT   READYMACHINECOUNT   UPDATEDMACHINECOUNT   DEGRADEDMACHINECOUNT   AGE
master   rendered-master-d9a03f612a432095dcde6dcf44597d90   True      False      False      3              3                   3                     0                      31h
worker   rendered-worker-f6827a4efe21e155c25c21b43c46f65e   True      False      False      6              6                   6                     0                      31h
```

----
+
** As the cluster owner, the following command displays the proxy status:
+
[source,terminal]
----
$ oc get proxy cluster -o yaml
----
+
.Sample Output
[source,terminal]
----
apiVersion: config.openshift.io/v1
kind: Proxy
spec:
  httpProxy: http://proxy.host.domain:8080
  httpsProxy: https://proxy.host.domain:8080
  <...more...>
status:
  httpProxy: http://proxy.host.domain:8080
  httpsProxy: https://proxy.host.domain:8080
  <...more...>

> ----

Reviewer 2: @EricPonvelle It would be better to replace "8080" to port.

// Module included in the following assemblies:
//
// * networking/configuring-cluster-wide-proxy.adoc

//OSDOCS-3001 Configuration for cluster-wide-proxy Day 2
:_content-type: PROCEDURE
[id="cluster-wide-proxy-updates_{context}"]
= Configuring or updating your cluster-wide proxy after installation

ifdef::openshift-rosa[]
As the cluster owner, you may wish to add a proxy to your created cluster after installation, or you may wish to make changes to your proxy that you configured during cluster installation. The `rosa` CLI provides some options for adding a proxy to your cluster or modifying an existing proxy on your cluster.
endif::[]
ifdef::openshift-dedicated[]
As the cluster owner, you may wish to add a proxy to your created cluster after installation, or you may wish to make changes to your proxy that you configured during cluster installation. The `ocm` CLI provides some options for adding a proxy to your cluster or modifying an existing proxy on your cluster.
endif::[]

You may need to perform these actions if:

* the cluster-wide proxy is configured after installation,

> * the proxy's network address needs to be updated,

Reviewer 3: Do you need and/or in th2 next to last bullet?
